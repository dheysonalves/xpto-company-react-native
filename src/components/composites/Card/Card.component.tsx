import React from "react";
import { View, Text, StyleSheet, Image, TouchableOpacity } from "react-native";
import Formatter from "../../../utils/Formatter";

interface ICardProps {
	data: {
		id?: string;
		name: string;
		cnpj: string;
		description: string;
		logo: string;
		createdAt?: string;
	};
	onHandleSubmit?: () => void;
}

const Card = ({ data, onHandleSubmit }: ICardProps) => {
	return (
		<TouchableOpacity
			style={styles.container}
			onPress={onHandleSubmit}
			activeOpacity={0.5}>
			<Image
				style={styles.image}
				source={{
					uri: Formatter.isImage(data.logo)
						? data.logo
						: "https://banqi.com.br/assets/img/uploads/img-app.png",
				}}
				height={52}
				width={52}
			/>
			<View style={styles.content}>
				<View style={styles.topContent}>
					<View>
						<Text style={[styles.title]} testID="cnpjText">
							CNPJ
						</Text>
						<Text style={styles.text}>{Formatter.formatCNPJ(data.cnpj)}</Text>
					</View>
					<View>
						<Text style={[styles.title]} testID="createdAtText">
							Created At
						</Text>
						<Text style={styles.text}>
							{Formatter.formatDate(data.createdAt as string)}
						</Text>
					</View>
				</View>
				<View style={styles.bottomContent}>
					<View style={styles.nameWrapper}>
						<Text style={[styles.title]} testID="nameText">Name</Text>
						<Text style={[styles.text]}>{data.name}</Text>
					</View>
					<View>
						<Text style={[styles.title]} testID="descriptionText">Description</Text>
						<Text style={styles.text}>
							{Formatter.removeWidowWord(data.description)}
						</Text>
					</View>
				</View>
			</View>
		</TouchableOpacity>
	);
};

const styles = StyleSheet.create({
	container: {
		flexDirection: "row",
		justifyContent: "flex-start",
		alignItems: "center",
		borderRadius: 5,
		borderColor: "#000",
		borderWidth: 2,
		padding: 16,
		marginVertical: 16,
	},
	image: {
		borderRadius: 50,
		width: 64,
		height: 64,
		marginRight: 16,
	},
	content: {
		flex: 1,
		flexDirection: "column",
		justifyContent: "space-between",
		alignItems: "center",
	},
	topContent: {
		width: "100%",
		flexDirection: "row",
		justifyContent: "space-between",
		alignItems: "center",
		paddingBottom: 12,
	},
	bottomContent: {
		width: "100%",
		flexDirection: "column",
		justifyContent: "space-between",
		alignItems: "flex-start",
	},
	text: {
		fontSize: 16,
	},
	title: {
		color: "#0a0a0a",
		fontWeight: "500",
		fontSize: 14,
		textTransform: "uppercase",
	},
	nameWrapper: {
		paddingBottom: 8,
	},
});

export default Card;
