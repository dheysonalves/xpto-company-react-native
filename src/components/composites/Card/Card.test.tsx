import React from "react";

import renderer from "react-test-renderer";
import {
	render,
} from "@testing-library/react-native";
import Card from "./Card.component";
import Companies from "../../../mocks/companies";

const CardRender = () => {
	return <Card data={Companies[0]} onHandleSubmit={() => {}} />;
};

describe("Card component test suite", () => {
	it("should return the Card component", () => {
		const tree = renderer.create(<CardRender />).toJSON();

		expect(tree).toMatchSnapshot();
	});

	it("should render the component texts", async () => {
		const cnpjText = "CNPJ";
		const createdAtText = "Created At";
		const nameText = "nameText";
		const descriptionText = "descriptionText";

		const { getByText, getByTestId, queryByTestId } = render(<CardRender />);

		const cnpj = getByText(cnpjText);
		const createdAt = getByText(cnpjText);
		const name = getByText(cnpjText);
		const description = getByText(cnpjText);


		expect(cnpj.props.children).toEqual(cnpjText);
		expect(createdAt.props.children).toEqual(cnpjText);
		expect(name.props.children).toEqual(cnpjText);
		expect(description.props.children).toEqual(cnpjText);
	});
});
