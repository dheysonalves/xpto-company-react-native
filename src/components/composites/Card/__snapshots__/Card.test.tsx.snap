// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component test suite should return the Card component 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#000",
      "borderRadius": 5,
      "borderWidth": 2,
      "flexDirection": "row",
      "justifyContent": "flex-start",
      "marginVertical": 16,
      "opacity": 1,
      "padding": 16,
    }
  }
>
  <Image
    height={52}
    source={
      Object {
        "uri": "https://banqi.com.br/assets/img/uploads/img-app.png",
      }
    }
    style={
      Object {
        "borderRadius": 50,
        "height": 64,
        "marginRight": 16,
        "width": 64,
      }
    }
    width={52}
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 12,
          "width": "100%",
        }
      }
    >
      <View>
        <Text
          style={
            Array [
              Object {
                "color": "#0a0a0a",
                "fontSize": 14,
                "fontWeight": "500",
                "textTransform": "uppercase",
              },
            ]
          }
          testID="cnpjText"
        >
          CNPJ
        </Text>
        <Text
          style={
            Object {
              "fontSize": 16,
            }
          }
        >
          84.649.484/9464-64
        </Text>
      </View>
      <View>
        <Text
          style={
            Array [
              Object {
                "color": "#0a0a0a",
                "fontSize": 14,
                "fontWeight": "500",
                "textTransform": "uppercase",
              },
            ]
          }
          testID="createdAtText"
        >
          Created At
        </Text>
        <Text
          style={
            Object {
              "fontSize": 16,
            }
          }
        >
          31/05/2022
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flexDirection": "column",
          "justifyContent": "space-between",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "paddingBottom": 8,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#0a0a0a",
                "fontSize": 14,
                "fontWeight": "500",
                "textTransform": "uppercase",
              },
            ]
          }
          testID="nameText"
        >
          Name
        </Text>
        <Text
          style={
            Array [
              Object {
                "fontSize": 16,
              },
            ]
          }
        >
          Testing
        </Text>
      </View>
      <View>
        <Text
          style={
            Array [
              Object {
                "color": "#0a0a0a",
                "fontSize": 14,
                "fontWeight": "500",
                "textTransform": "uppercase",
              },
            ]
          }
          testID="descriptionText"
        >
          Description
        </Text>
        <Text
          style={
            Object {
              "fontSize": 16,
            }
          }
        >
          MEI
        </Text>
      </View>
    </View>
  </View>
</View>
`;
